pkg/sharing/replicator.go

```diff
@@ -0,0 +1,190 @@
+package sharing
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/cozy/cozy-stack/client/request"
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/instance"
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+// ReplicateMsg is used for jobs on the share-replicate worker.
+type ReplicateMsg struct {
+	SharingID string `json:"sharing_id"`
+}
+
+// Replicate starts a replicator on this sharing.
+func (s *Sharing) Replicate(inst *instance.Instance) error {
+	if !s.Owner {
+		return s.ReplicateTo(inst, &s.Members[0])
+	}
+	var errm error
+	for i, m := range s.Members {
+		if i == 0 {
+			continue
+		}
+		if m.Status == MemberStatusReady {
+			err := s.ReplicateTo(inst, &s.Members[i])
+			errm = multierror.Append(errm, err)
+		}
+	}
+	return errm
+}
+
+// ReplicateTo starts a replicator on this sharing to the given member.
+// See http://docs.couchdb.org/en/2.1.1/replication/protocol.html
+func (s *Sharing) ReplicateTo(inst *instance.Instance, m *Member) error {
+	if m.Instance == "" {
+		return ErrInvalidURL
+	}
+
+	// TODO get the last sequence number
+
+	changes, err := s.callChangesFeed(inst)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("changes = %#v\n", changes)
```

Comment:
`fmt.Printf` to remove

Reply:
Or maybe you're doing because its a WIP, and dont want the compiler to yell ?

Reply:
Yes, this function is a WIP